File: frontend/src/types/route.ts
export interface RouteRecord {
  id: number | null;
  title: string;
  description: string;
  qrUrl: string;
  coverUrl?: string;
  createTime?: string;
  isDeleted?: number;
  userId?: number;
  status?: number | null;
}

export interface RouteOverview extends RouteRecord {
  authorName: string;
  authorAvatar: string;
  favoritesCount: number;
  favorited: boolean;
}

export interface RouteListPayload {
  page: number;
  pageSize?: number;
  keyword?: string;
  userId?: number;
  startTime?: string;
  endTime?: string;
}

export interface AddRoutePayload {
  title: string;
  description: string;
  qrUrl: string;
  coverUrl?: string;
  userId: number;
}

export interface UpdateRoutePayload {
  id: number;
  title?: string;
  description?: string;
  qrUrl?: string;
  coverUrl?: string;
  status?: number | null;
}

export interface DeleteRoutesPayload {
  ids: number[];
}

export interface PageResult<T> {
  total: number;
  records: T[];
}

File: backend/QRForRoutes/src/main/java/com/liu/qrRoutes/model/Route.java
package com.liu.qrRoutes.model;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;

@Data
@TableName("route")
public class Route implements Serializable {
  private Integer id;
  private String title;
  private String description;
  private String qrUrl;
  private String coverUrl;
  private String createTime;
  private Byte isDeleted;
  private Integer userId;
  private Byte status;

  @TableField(exist = false)
  private Long favoritesCount;

  @TableField(exist = false)
  private Boolean favorited;
}
